content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,137 @@
+---
+page_title: Hooks - Terraform Cloud Agents - Terraform Cloud and Terraform Enterprise
+---
+
+# Hooks
+
+Terraform Cloud Agents support executing custom programs, or hooks, during
+strategic points of a Terraform run. These hooks allow operators to extend the
+functionality of Terraform runs.
+
+Terraform Cloud Agents support the following hooks:
+
+- `pre-plan` - This hook executes before `terraform init` during a `plan`
+operation.
+
+- `post-plan` - This hook executes after `terraform plan` during a `plan`
+operation regardless whether or not `terraform plan` successfully executed.
+
+- `pre-apply` - This hook executes before `terraform init` during an `apply`
+operation.
+
+- `post-apply` - This hook executes after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` successfully executed.
+
+## Limitations
```

Comment:
I'm wondering if a different title is better here. It's just that a lot of them don't feel like limitations exactly, they feel like requirements. Thoughts on this?

Reply:
Updated it to `Hook Behavior`.